Sources/NIO/ConvenienceOptionSupport.swift

```diff
@@ -0,0 +1,421 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// MARK: ServerBootstrap - Server
+extension ServerBootstrap {
+    /// Specifies some `ChannelOption`s to be applied to the `ServerSocketChannel`.
+    /// - See: serverChannelOption
+    /// - Parameter options: List of shorthand options to apply.
+    /// - Returns: The updated server bootstrap (`self` being mutated)
+    @inlinable
+    public func serverChannelOptions(_ options: [NIOTCPServerShorthandOption]) -> ServerBootstrap {
+        var toReturn = self
+        for option in options {
+            toReturn = toReturn.serverChannelOption(option)
+        }
+        return toReturn
+    }
+    
+    @usableFromInline
+    func serverChannelOption(_ option: NIOTCPServerShorthandOption) -> ServerBootstrap {
+        let applier = ServerBootstrapServer_Applier(contained: self)
+        return option.applyOptionDefaultMapping(applier).contained
+    }
+    
+    fileprivate struct ServerBootstrapServer_Applier: NIOChannelOptionAppliable {
```

Comment:
It's a bit sad that we have to have so many of these. Each one is basically a copy-and-replace of the same shape. Is there any concrete reason we can't replace all of these with a closure, instead of a struct and generics? That's essentially what they are, but with a lot more duplication.

Reply:
@Lukasa / @weissi  - I've reworked using an idea Weissi of turning the array into a structure as soon as possible.  The options can be consumed by bootstraps now - mostly to help universal underlying do it right.

We could make it a runtime error to not consume all options in the underlying (ie, force them to call applyFallbackMapping themselves.  This I think would allow us to do everything interns of a ChannelOptions.Storage rather than using the applier structure.

Reply:
Having a return type of `Self` is problematic which ultimately comes from the bootstrap method of `public func serverChannelInitializer(_ initializer: @escaping (Channel) -> EventLoopFuture<Void>) -> Self` and friends.  Not sure how to capture that concept in a closure.  Maybe there is some way where it's Called with a parameter generic on the type of bootstrap you want back or something similar?

Reply:
Actually - it's coming back to be now.  The bigger problem is that the end usage is generic on the type of ChannelOption.  So it's representing effectively a 'generic closure'